File: code/java/validPalindromeII.java
class Solution {
    public boolean validPalindromeII(String s) {
        int left = 0;
        int right = s.length() - 1;
        while(left<right){
            if(s.charAt(left) == s.charAt(right)){
                left ++;
                right --;
            }
            else
                return (helper(s.substring(left,right)) || helper(s.substring(left+1,right+1)));
        }
        return true;
    }
    
    public boolean helper(String s){
        int left = 0;
        int right = s.length() - 1;
        while(left<right)
            if(s.charAt(left++) != s.charAt(right--))
                return false;
        return true;
    }
}

File: code/python/validPalindromeII.py
class Solution:
    def validPalindromeII(self, s: str) -> bool:
        left = 0
        right = len(s)-1
        while left < right:
            if s[left] == s[right]:
                left += 1
                right -= 1
            else:
                return (self.helper(s[left:right]) or self.helper(s[left+1:right+1]))

        return True
    
    
    def helper(self,s:str) -> bool:
        le = 0
        ri = len(s) - 1
        while le < ri:
            if s[le] != s[ri]:
                return False
            else:
                le += 1
                ri -= 1
                
        return True
